src/Connection.js

```diff
@@ -20,8 +20,9 @@ export class Connection {
   }
 
   start({sessionCreator, dbHost, dbPort, dbAuthKey, dbSsl}) {
-    const urlQueryParams = url.parse(this.webSocket.upgradeReq.url, true).query;
-    this.sessionPromise = sessionCreator(urlQueryParams).catch(e => {
+    const urlQueryParams = url.parse(this.webSocket.upgradeReq.url, true).query,
+          headers        = this.webSocket.upgradeReq.headers;
```

Comment:
This is failing eslint style check, can you rewrite as `const headers = this.webSocket.upgradeReq.headers;`?

And I'm wondering if we may as well be more general and pass the entire `upgradeReq` object instead of just `headers`, and let the developer get whatever they want out of it. There may be other useful attributes besides the query params and the headers. Or is that too much power, and could allow a developer to use it in the wrong way like sending a response?